src/main/native/com/code_intelligence/jazzer/driver/fuzzed_data_provider.cpp

```diff
@@ -96,9 +99,14 @@ ConsumeIntegralArray(JNIEnv &env, jobject self, jint max_length) {
   const auto *dataPtr =
       reinterpret_cast<const uint8_t *>(env.GetLongField(self, gDataPtrField));
   jint remainingBytes = env.GetIntField(self, gRemainingBytesField);
-
+  uint64_t requested_bytes = sizeof(T) * max_length;
+  // cap to max int32 so we don't request negative bytes
+  uint64_t capped_bytes =
+      std::min(requested_bytes,
+               static_cast<uint64_t>(std::numeric_limits<int32_t>::max()));
   jint max_num_bytes =
-      std::min(static_cast<jint>(sizeof(T)) * max_length, remainingBytes);
+      std::min(static_cast<jint>(capped_bytes), remainingBytes);
+
```

Comment:
It seems easier to just cap to `static_cast<uint32_t>(remaining_bytes)` directly, then cast to `jint`. The separate cap to `std::numeric_limits<int32_t>::max()` should be unnecessary as `remaining_bytes` is always at most that large.